chore: Installation script improvements (#7433)
SAP/spartacus: scripts/install/README.md

- `ANGULAR_CLI_VERSION` variable is set properly in the `config.sh` file (it must meet project's `package.json` `@angular/cli` version requirement)
- `OCC_PREFIX` variable is set to proper value (depends on backend version)
- adjust any other variables available in the `config.default.sh` if needed (by setting them up in the `config.sh`)

Once ready start the installation as below:

```bash
 ./run.sh install
```

The script will build the libraries from source, create a new shell app and install the libraries in it.

By default, Spartacus will be built from the `develop` branch. You can specify a different branch in the `config.sh` file.

The installation will create two folders in the `$BASE_DIR` directory:

- `apps` (generated/executable apps)
- `clone` (cloned Spartacus repo)

## Starting Spartacus

```bash
./run.sh start
```

This command starts a server for Spartacus, which is under `"${BASE_DIR}/apps/csr"` by default.

## Stop Spartacus

```bash
./run.sh stop
```

This command stops the running Spartacus PM2 processes.

## Update Spartacus

Just run the installation again (stop it first if it's running).

```bash
./run.sh stop

> ./run.sh install (script ask you to rename/delete the `$BASE_DIR` if it exists)

Reviewer: It will ask or it just removes the directory?

Author: It will remove, it's the old message.

# Installation Script for Spartacus

Utilities and scripts to automate the Spartacus installation steps. Some of them are:

- Build Spartacus libraries from source
- Create a new shell app for the Spartacus libraries
- Install the libraries in the new shell app
- Add custom configurations to the shell app
- Install spartacus in production mode
- Install Spartacus in SSR mode
- Install Spartacus in remote servers (using Ansible)

To use the scripts, clone this git repo and follow the usage indications below.

## Install Spartacus

Script's output directory is set as below (related to script's root, `./scripts/install`):
```bash
# top directory for the installation output (must be outside of the project)
if [ -z $BASE_DIR ]; then
    BASE_DIR="../../../spartacus-${SPARTACUS_VERSION}"
fi
```
so it's one level above the spartacus project root. To change the output directory `$BASE_DIR` can be overriden in the `config.sh` file.

> All scirpt's commands should be run while being in it's root directory.

Reviewer: ```suggestion
All script's commands should be run while being in it's root directory.
```

```bash
cd ./scripts/install && ./run.sh install
```

The script will build the libraries from source, create a new shell app and install the libraries in it.

By default, Spartacus will be built from the `develop` branch. You can specify a different branch in the `config.sh` file.

The installation will create two folders in the `$BASE_DIR` directory:

- `apps` (generated/executable apps)
- `clone` (cloned Spartacus repo)

## Starting Spartacus

```bash
./run.sh start
```

> This command starts a development server for Spartacus, which is under `"${BASE_DIR}/apps/csr"` by default.

Reviewer: ```suggestion
This command starts a server for Spartacus, which is under `"${BASE_DIR}/apps/csr"` by default.
```

## Stop Spartacus

```bash
./run.sh stop
```

> This command stops the running Spartacus using its PID.

Reviewer: ```suggestion
This command stops the running Spartacus PM2 processes.
```

> # Spartacus Installation Script

Reviewer: Could we also update the docs in this PR?
